src/libpmemobj/critnib.c

```diff
@@ -516,13 +502,18 @@ critnib_remove(struct critnib *c, uint64_t key)
 	ASSERTne(ochild, -1);
 
 	store(n_parent, n->child[ochild]);
-	void *value = k->value;
 	c->pending_del_nodes[del] = n;
+
+del_leaf:;
+	void *value = k->value;
 	c->pending_del_leaves[del] = k;
 
```

Comment:
put an `out:` label here, and then make sure that value is always NULL when you call it (which should be already true).